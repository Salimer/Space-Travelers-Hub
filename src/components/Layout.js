import { Outlet } from 'react-router-dom';
import NavBar from './NavBar';

const Layout = () => (
  <>
    <NavBar />
    <Outlet />
  </>
);

export default Layout;
