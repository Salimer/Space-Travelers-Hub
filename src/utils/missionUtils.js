const filterReserved = (missions) => missions.filter((mission) => mission.reserved);

export default filterReserved;
