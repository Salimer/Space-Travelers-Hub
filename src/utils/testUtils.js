import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

const withRouter = (component) => (
  <BrowserRouter>
    {component}
  </BrowserRouter>
);

const withProvider = (component) => (
  <Provider store={store}>
    {component}
  </Provider>
);

export { withProvider, withRouter };
