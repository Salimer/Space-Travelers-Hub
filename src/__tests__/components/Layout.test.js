import renderer from 'react-test-renderer';
import { withRouter } from '../../utils/testUtils';
import Layout from '../../components/Layout';

describe('testing Layout component', () => {
  test('render', () => {
    const tree = renderer.create(withRouter(<Layout />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
