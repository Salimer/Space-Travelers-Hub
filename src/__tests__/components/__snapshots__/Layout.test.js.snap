// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Layout component render 1`] = `
Array [
  <nav
    className="sc-bgqQcB jQzTWm mb-4 py-3"
  >
    <div
      className="container"
    >
      <div
        className="justify-content-between align-items-center hstack"
      >
        <div
          className="hstack gap-3"
        >
          <img
            className=""
            src="planet.png"
            style={
              Object {
                "height": "3em",
                "width": "3em",
              }
            }
          />
          <h3>
            Space Traveler's Hub
          </h3>
        </div>
        <ul
          className="hstack gap-4"
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <li>
            <a
              aria-current="page"
              className="active"
              href="/"
              onClick={[Function]}
            >
              rockets
            </a>
          </li>
          <li>
            <a
              className=""
              href="/missions"
              onClick={[Function]}
            >
              missions
            </a>
          </li>
          <div
            className="sc-gTRrQi hudrtB vr"
          />
          <li>
            <a
              className=""
              href="/my-profile"
              onClick={[Function]}
            >
              my profile
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  />,
]
`;
